### PR TITLE
feat: replace comunity-lib with kafka-lib

### PR DESCRIPTION
**Pull request by VictorZhou123**

### CODE REVIEW

File: common/infrastructure/redis/config.go [hunk 1]
```diff
@@ -6,6 +6,8 @@ type Config struct {
 	Address  string `json:"address"   required:"true"`
 	Password string `json:"password"  required:"true"`
 	KeyPair  string `json:"key_pair"  required:"true"`
+	DB       int    `json:"db"`
```
Comment (zengchen1024) [inline common/infrastructure/redis/config.go:9]: use redis-lib instead.
使用redis-lib来替换这个redis的封装

File: message-server/config.go [hunk 2]
```diff
@@ -36,9 +39,24 @@ type PostgresqlConfig struct {
 	asyncconf asyncrepoimpl.Config
 }
 
-func (cfg *configuration) getMQConfig() mq.MQConfig {
-	return mq.MQConfig{
-		Addresses: cfg.MQ.ParseAddress(),
+// func (cfg *configuration) getMQConfig() mq.MQConfig {
```
Comment (zengchen1024) [inline message-server/config.go:42]: delete this comments 删除这些注释

File: message-server/main.go [hunk 3]
```diff
@@ -82,11 +84,17 @@ func main() {
 	}
 
 	// mq
-	if err := messages.Init(cfg.getMQConfig(), log, cfg.MQ.Topics); err != nil {
+	kfkCfg := cfg.getKfkConfig()
+	redisCfg := cfg.getRedisConfig()
+	err = redislib.Init(&redisCfg)
+	if err != nil {
+		log.Fatalf("initialize redis of kafka failed, err:%v", err)
+	}
```
Comment (zengchen1024) [inline message-server/main.go:92]: defer redislib.Close()

File: infrastructure/messages/subscribe.go [hunk 4]
```diff
@@ -17,143 +16,91 @@ import (
...
+	handlerNameAddFollowing       = "add_following"
+	handlerNameAddLike            = "add_like"
+	handlerNameAddFork            = "add_fork"
+	handlerNameAddDownload        = "add_download"
+	handlerNameAddRelatedResource = "add_related_resource"
+	handlerNameCreateTraining     = "create_training"
+	handlerNameCreateFinetune     = "create_finetune"
+	handlerNameCreateInference    = "create_inference"
+	handlerNameCreateEvaluate     = "create_evaluate"
+	handlerNameCreateCloud        = "create_cloud"
```
Comment (zengchen1024) [inline infrastructure/messages/subscribe.go:30]: format these constant variables. 对齐这些常量，create的放在一起，add的放在一起，这样才能看清楚是否有重复定义。

File: infrastructure/messages/subscribe.go [hunk 5]
```diff
@@ -264,264 +196,241 @@ func registerHandlerForDownload(handler interface{}) (mq.Subscriber, error) {
...
-			return
-		}
+func registerHandlerForBigModel(handler interface{}) error {
 
-		body := bigmoddelmsg.MsgTask{}
-		if err = json.Unmarshal(msg.Body, &body); err != nil {
-			return
-		}
+	return subscribe(topics.BigModel, handlerNameCreateBigModel,
+		func(b []byte, m map[string]string) (err error) {
```
Comment (zengchen1024) [inline infrastructure/messages/subscribe.go:398]: f:=func(b []byte, m map[string]string) (err error) {}
return subscribe(topics.BigModel, handlerNameCreateBigModel, f)

**Review by zengchen1024 — CHANGES_REQUESTED** [review]

File: common/infrastructure/redis/config.go (hunk 1 above)
Comment (VictorZhou123) [reply common/infrastructure/redis/config.go:9]: 后续集成到redis-lib

File: message-server/config.go (hunk 2 above)
Comment (VictorZhou123) [reply message-server/config.go]: done

File: message-server/main.go (hunk 3 above)
Comment (VictorZhou123) [reply message-server/main.go:92]: 后面的commit中有这部分redis和kafka初始化的封装

File: infrastructure/messages/subscribe.go (hunk 4 above)
Comment (VictorZhou123) [reply infrastructure/messages/subscribe.go:30]: done

File: infrastructure/messages/subscribe.go (hunk 5 above)
Comment (VictorZhou123) [reply infrastructure/messages/subscribe.go]: done

File: infrastructure/messages/subscribe.go [hunk 6]
```diff
@@ -290,23 +217,21 @@ func registerHandlerForRelatedResource(handler interface{}) (mq.Subscriber, erro
 		}
 
 		return nil
-	})
+	}
+
+	return subscribe(
+		topics.RelatedResource, handlerNameAddRelatedResource, f)
```
Comment (zengchen1024) [inline infrastructure/messages/subscribe.go:223]: return subscribe(
topics.RelatedResource, handlerNameAddRelatedResource, f,
)

File: infrastructure/messages/subscribe.go [hunk 7]
```diff
@@ -522,6 +431,13 @@ func registerHandlerForBigModel(handler interface{}) (mq.Subscriber, error) {
...
 		return
+	}
 
-	})
+	return subscribe(topics.BigModel, handlerNameCreateBigModel, f)
+}
+
+func subscribe(topicName string, handlerName string, handler kfklib.Handler) error {
+	return kfklib.SubscribeWithStrategyOfRetry(
+		handlerName, handler, []string{topicName}, 3,
```
Comment (zengchen1024) [inline infrastructure/messages/subscribe.go:441]: const retryNum = 3

File: main.go [hunk 8]
```diff
@@ -100,11 +100,15 @@ func main() {
 	}
 
 	// mq
-	if err := messages.Init(cfg.GetMQConfig(), log, cfg.MQ.Topics); err != nil {
+	if err = messages.InitKfkLib(
```
Comment (zengchen1024) [inline main.go:103]: It is better to pass the redlis instance. because it will not couple with kafka

**Review by zengchen1024 — CHANGES_REQUESTED** [review]

**Pull request merged**